src/usecases/addCourseByManual.ts

```diff
@@ -0,0 +1,39 @@
+import { usePorts } from "~/usecases/index";
+import { Methods } from "~/api/registered-courses/index";
+import { RegisteredCourseWithoutID } from "~/api/@types";
+import { Ports } from "~/adapter";
+
+export const formatCourse = (
+  course: RegisteredCourseWithoutID
+): Methods["post"]["reqBody"] => ({
+  year: course.year,
+  name: course.name ?? "",
+  instructor: course.instructor ?? "",
+  credit: course.credit ?? -1,
+  methods: course.methods ?? [],
+  schedules: course.schedules ?? [],
+  tags: course.tags,
+});
+
+/**
+ * 講義情報をサーバに登録し、成功すれば Vuex にも登録する。
+ */
+export const addCourseByManual = ({ api, store }: Ports) => async (
+  course: RegisteredCourseWithoutID
+) => {
+  console.dir(course);
+  try {
+    const {
+      body: registeredCourse,
+      status: status,
+    } = await api.registered_courses.post({
+      body: formatCourse(course),
+    });
+    if (status !== 200) throw "Result is NOT 200";
+    store.commit("addCourse", registeredCourse);
+    return registeredCourse;
+  } catch (error) {
+    console.error(error);
+    return false;
```

Comment:
return typeがややこしくなっちゃうのでnullにしたい。
```suggestion
    return null;
```

Reply:
csvでやってたvoid返すやつでもいいかもね

Reply:
いいですね。ここは`void 0`にします。
`null`と`undefined`の扱いは統一したい感はありますよね。`void 0` = `undedined`で、さらに`null == undefined`は`true`になるので、ここは`null`でも`void 0`でもどっちでも動作は変わりません。変わらないのに`null`と`undefined`の２つの型があるのは混乱します。
[Micorosft のガイドライン](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined)では`null` は使わず `undefined` に統一すると書いてあったりします。

Reply:
undefinedはフロント側が定義してないような値にのみ使われるのでreturn undefinedとかはしないほうがよく、nullは値が存在しないことを表すので、正常な処理をおこなったが値はないときとかに使う。return  undefinedになりそうなときはreturn nullの方が直感的。void 0はundefinedと同じだか、voidを返すときに利用するのはいいかも（自分はvoidは返すぐらいなら例外をcatchする派）↑の文章は個人の見解です

ちなみにbottom typeのneverもいる

これを分けるのも、逆に統一するのも面倒なので、サーバー見ながらeslintを調整してもいいかも。

Reply:
結論としてはここは`void 0`でいいのでしょうか？また、
> void 0はundefinedと同じだか、voidを返すときに利用するのはいいかも

「voidを返すときに利用するのはいい」の意味が汲み取れませんでした… `void` == `undefined` なのになぜ`void`を返すときには利用していいのでしょうか？

Reply:
> 結論としてはここはvoid 0でいいのでしょうか？

:ok:

> void == undefined なのになぜ「voidを返すとき」には利用していいのでしょうか？

1. void 0 読みにくくて、書き方の選択肢が増えるので、できるだけ減らしたいモチベ
2. 型の一致が気になる人もいると思うので void 0 は場合によって使ってもいいよ
3. 文字数を減らすために undefined を void 0 にするといった理由だと分かりにくくなると感じたので利用を限定して書きました